### PR TITLE
Fixes for executable documents

### LINKED ISSUE #1086: Executable documents are broken
The latex embedding was not using the new parameters.

It also looks like the generation of the images are not working.

### PR DESCRIPTION
**Pull request by rexissimus**
These are the fixes required for local executable documents to work.

* Adds noExecution command-line option
* Temporarily replaces temp_configuration when executing commands from other instance. (temp_configuration=persistent+command-line flags).
* Disables IPython mode when executing commands from other instance, because it thinks the vistrails console is the active ipython console?
* latex code now uses output modules (But no PDF option for VTK files!)
* Added test script that compiles and checks size of final pdf. This should be set up on a buildbot.

Fixes #1086

### CODE REVIEW

File: vistrails/gui/application.py [hunk 1]
```diff
@@ -524,10 +524,14 @@ def noninteractiveMode(self):
...
+            if self.temp_configuration.check('outputDirectory'):
+                output_dir = self.temp_configuration.outputDirectory
+            else:
+                output_dir = None
+
             errs = []
-            if self.temp_configuration.check('workflowGraph'):
-                workflow_graph = self.temp_configuration.workflowGraph
-                results = vistrails.core.console_mode.get_wf_graph(w_list, workflow_graph,
+            if self.temp_configuration.check('withWorkflow'):
```
Comment (remram44) [inline vistrails/gui/application.py]: Maybe this should be renamed, ex "--output-version-tree"

Comment (rexissimus) [inline vistrails/gui/application.py]: Yes, it does seem clearer to use `output` prefix.
Also, use correct `graph` and `tree` descriptions?

    withWorkflow -> outputPipelineGraph
    withVersionTree -> outputVersionTree

@dakoop what do you think?

File: vistrails/gui/application.py [hunk 2]
```diff
@@ -769,21 +765,32 @@ def parse_input_args_from_other_instance(self, msg):
...
-                        # so builderWindow is activated
-                        self.builderWindow.raise_()
-                        self.builderWindow.activateWindow()
-                    return result
-                else:
-                    return self.noninteractiveMode()
+                try:
+                    # Execute using persistent configuration + new temp configuration
+                    old_temp_conf = self.temp_configuration
+                    self.startup.temp_configuration = self.configuration.__copy__()
```
Comment (remram44) [inline vistrails/gui/application.py]: This should be copy()

**Comment by remram44** [comment]
So just to be clear, a VisTrails instance is going to get spawned for every figure in the LaTeX file if one is not currently running? Else the currently running instance is going to get them all through the socket?

I would be a lot easier if a new instance was always spawned when using batch mode, but it would be really slow if there are lots of figures and they use something like VTK (whose initialization makes for a very noticeable runtime). On the other hand, if this is very slow unless you start the VisTrails UI first, it is really awkward as well.

**Comment by rexissimus** [comment]
Yes. The main reason for using a running instance is that it is faster. This is very useful in batch mode so you do not have to initialize vistrails for each executed workflow. This would also be useful when using the vistrails API in python scripts, but we would need tighter integration with the running instance, and it would only work on serializeable data.

You are right that this can get complicated and lead to issues with the execution environment, but I don't see those as big problems.

Using a GUI instance is awkward. We could have a daemon-style instance that starts automatically and stays for some set time. The GUI could run on top of this and keep the daemon alive. But this would probably need a redesign of VisTrails.

We also have the "spawned" mode. It works differently, but is probably more suitable for tighter integration.

File: vistrails/gui/application.py (hunk 2 above)
Comment (rexissimus) [inline vistrails/gui/application.py]: fixed

**Pull request merged**